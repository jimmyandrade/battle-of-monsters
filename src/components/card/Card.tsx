import type { CardProps } from './Card.types'
import type { FC } from 'react'
import { CardRoot } from './Card.styles'

export const Card: FC<CardProps> = ({
  component = 'div',
  className,
  id,
  ...props
}) => (
  <CardRoot as={component} className={className} id={id} {...props} />
)
