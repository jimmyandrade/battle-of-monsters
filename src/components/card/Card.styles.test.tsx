import { render } from '@testing-library/react'
import { Card } from './Card'

describe('Card styles', () => {
  describe('Root component', () => {
    it('should match snapshot', () => {
      const { container } = render(<Card />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
