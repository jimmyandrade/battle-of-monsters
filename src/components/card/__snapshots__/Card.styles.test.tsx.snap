// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card styles Root component should match snapshot 1`] = `
.c0 {
  background-color: #FFFFFF;
  border: 1px solid transparent;
  border-radius: var(--card-border-radius);
  box-shadow: var(--box-shadow);
  display: block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

@media (prefers-reduced-motion:no-preference) {
  .c0 {
    -webkit-transition-duration: var(--micro-animation-duration);
    transition-duration: var(--micro-animation-duration);
    -webkit-transition-timing-function: ease-in-out;
    transition-timing-function: ease-in-out;
    -webkit-transition-property: border-color,box-shadow;
    transition-property: border-color,box-shadow;
  }
}

<div
  class="c0"
/>
`;
